### PR TITLE
Add rack support

### LINKED ISSUE #54: Make racks configurable
**Is your feature request related to a problem? Please describe.**
Users should be able to configure the racks of a `CassandraDatacenter`. Here is an example manifest:

```yaml
apiVersion: cassandra.datastax.com/v1beta1
kind: CassandraDatacenter
metadata:
  name: k8ssandra-test
spec:
  clusterName: cluster1
  serverType: cassandra
  serverVersion: "3.11.7"
  managementApiAuth:
    insecure: {}
  size: 3
  storageConfig:
    cassandraDataVolumeClaimSpec:
      storageClassName: server-storage
      accessModes:
        - ReadWriteOnce
      resources:
        requests:
          storage: 5Gi
  racks:
    - name: rack1
      zone: us-central1-a
    - name: rack2
      zone: us-central1-b
    - name: rack3
      zone: us-central1-c
```

**Describe the solution you'd like**
The `racks` property of `CassandraDatacenter` should support templated values in the [cassdc.yaml](https://github.com/k8ssandra/k8ssandra/blob/main/charts/k8ssandra-cluster/templates/cassdc.yaml) template.

### PR DESCRIPTION
**Pull request by jsanda**
Fixes #54.

#230 needs to be merged first.

The only thing I have left to do is update the comments in `k8ssandra/values.yaml` and bump the chart version.

### CODE REVIEW

**Comment by jsanda** [comment]
This is ready for review. I am particularly interested in any thoughts or suggestions around the formatting of the comment I have for the `racks` property in `k8ssandra/values.yaml`. Currently there are a couple summary lines followed by the property name commented out. A couple lines after that has a longer, much more detailed explanation with an example showing how to use it. I expect that some of this same info will go in the site docs, but I think it should also doc'd in `values.yaml`.

File: charts/k8ssandra/values.yaml [hunk 1]
```diff
@@ -40,6 +40,36 @@ cassandra:
...
+    # with a 9 node cluster and 3 racks (and 3 nodes per rack). There will be one replica
+    # per rack.
+    #
+    # Each element in the racks array consists of two properties - name and affinityLabels.
+    # name specifies the rack name and is required. affinityLabels are an optional set of
+    # labels that are used to pin Cassandra pods to specific k8s worker nodes via affinity
+    # rules. See https://kubernetes.io/docs/tasks/configure-pod-container/assign-pods-nodes-using-node-affinity/
+    # for background on using affinity rules.
+    #
+    # topology.kubernetes.io/zone is a well-known k8s label used by cloud providers to
```
Comment (burmanm) [inline charts/k8ssandra/values.yaml:59]: This is excellent addition, it clarifies one use-case to certain users.

**Comment by jsanda** [comment]
Here is a snippet of my Cassandra pod's spec that shows the affinity rules that cass-operator generates:

```yaml
spec:
  affinity:
    podAntiAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
      - labelSelector:
          matchExpressions:
          - key: cassandra.datastax.com/cluster
            operator: Exists
          - key: cassandra.datastax.com/datacenter
            operator: Exists
          - key: cassandra.datastax.com/rack
            operator: Exists
        topologyKey: kubernetes.io/hostname
```

File: charts/k8ssandra/values.yaml (hunk 1 above)
Comment (jeffbanks) [reply charts/k8ssandra/values.yaml:63]: 👍 on the `values.yaml` docs.  Right where you need the information when making changes.

**Review by jeffbanks — APPROVED** [review]
LGTM

**Pull request merged**